### PR TITLE
Moving all provider slack channels under sig-cloud-provider.

### PR DESCRIPTION
**Pull request by cheftako**
Follow up from discussion on #4178 

sig cloud provider is the owning sig for all the cloud provider groups, fixing our slack configuration to reflect that.

/sig cloud-provider

### CODE REVIEW

File: communication/slack-config/sig-cloud-provider/config.yaml [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+  - name: provider-gcp
+    id: C6WB33KNJ
+  - name: provider-ibmcloud
+    id: C90BWDWCE
+  - name: provider-openstack
+    id: C0LSA3T7C
+  - name: provider-vsphere
+    id: C9PGCDKV5
+  - name: provider-centurylink
+    id: C0P8MQUM9
```
Comment (mrbobbytables) [inline communication/slack-config/sig-cloud-provider/config.yaml]: More so an FYI - You only need to provide the channel `id` if it is being renamed 👍 
...I should have mentioned that in the original PR when moving `sig-cloud-provider` over. 🙈 

 A small nit and I wouldn't consider it a blocker, but figure while moving them all over, might as well alpha sort?

Comment (cheftako) [reply communication/slack-config/sig-cloud-provider/config.yaml]: So the ids (other than centrurylinks) were transferred from the source. 
It wasn't clear to me if they had been renamed, so I thought it was safer to just leave them.

I went ahead and alpha sorted them

**Comment by mrbobbytables** [comment]
Thank you kindly @cheftako 
/approve

**Pull request merged**